mvp_onboarding.client/src/services/customerService.js

```diff
@@ -21,8 +21,13 @@ export const getCustomers = async (pageNumber, pageSize, sortColumn, sortDirecti
 // Get customer 
 export const getCustomer = async (id) => {
   try {
-    const response = await api.get(`/Customer/${id}`);
-    return response.data;
+    if (!(Number.isInteger(id) && id >0)) {
+      throw "Invalid customer id.";
+    }
```

Comment:
Remember our discussion before about  SOLID principles? As you can notice here, you have the same **if** conditions for customer, product, sale, and store. You can create a single function for this one. For example, you can create a **util** folder and create a new file called **validation.js** This file will have a new function that contains the **if** condition and will return a boolean value.

Then for your services you can just call this new function for example:
`const isIDValid = idValidation(id)`
and then, setup something like
`if (!isIdValid) {
 throw "Invalid customer id.";
 return;
}
then your other codes
`

This way you can have a single function that you can manage and maintain.

Reply:
makes sense Kester. I wrote the separate validation function and I'm calling it from all services that need it.
I see how we can simplify the code by removing the else after the throw like you mentioned, even the return is unnecessary (it gave me a code unreachable warning if I write it like that). I find it more readable to leave the else in place but understand it is unnecessary and can be simplified.